Makefile.pre.in

```diff
@@ -936,8 +937,8 @@ pybuilddir.txt: $(PYTHON_FOR_BUILD_DEPS)
 		exit 1 ; \
 	fi
 
-build-details.json: pybuilddir.txt
-	$(RUNSHARED) $(PYTHON_FOR_BUILD) $(srcdir)/Tools/build/generate-build-details.py `cat pybuilddir.txt`/build-details.json
+$(shell [ -f pybuilddir.txt ] && cat pybuilddir.txt)/build-details.json: pybuilddir.txt
```

Comment:
You cannot use `$(shell` and should probably forget it exists :)

It is a GNU-ism and will break on any other Make implementation, such as the default Make on BSD.

If cpython were to move to require GNU Make then there are many changes which could be made to this Makefile that would make it significantly smaller, but alas, compatibility...